### PR TITLE
turned inner methods into async ones

### PR DESCRIPTION
**Pull request by Dowwie**

### CODE REVIEW

File: src/session.rs [hunk 1]
```diff
@@ -220,7 +220,7 @@ impl Inner {
                     if let Some(cookie) = jar.signed(&self.key).get(&self.name) {
                         let value = cookie.value().to_owned();
                         let cachekey = (self.cache_keygen)(&cookie.value());
-                        return Either::Left(
+                        return 
                             self.addr.send(Command(resp_array!["GET", cachekey])).map(
```
Comment (fafhrd91) [inline src/session.rs]: There is no need in this map in async/await context

File: src/session.rs [hunk 2]
```diff
@@ -311,8 +311,8 @@ impl Inner {
 
         let state: HashMap<_, _> = state.collect();
         match serde_json::to_string(&state) {
-            Err(e) => Either::Left(err(e.into())),
-            Ok(body) => Either::Right(
+            Err(e) => Err(e.into()),
+            Ok(body) =>
                 self.addr
                     .send(Command(resp_array!["SET", cachekey, body, "EX", &self.ttl]))
```
Comment (fafhrd91) [inline src/session.rs:315]: Same as above

**Review by fafhrd91 — CHANGES_REQUESTED** [review]
I don’t understand purpose of this pr. If you change to async why you keep future combinators?

File: src/session.rs (hunk 1 above)
Comment (Dowwie) [reply src/session.rs]: > I don’t understand purpose of this pr. If you change to async why you keep future combinators?

Yeah.. I see what you mean.  I managed to remove all the map combinators and consequently the need for FutureExt, too, in latest commit.

File: src/session.rs (hunk 2 above)
Comment (Dowwie) [reply src/session.rs:315]: done.

**Comment by Dowwie** [comment]
@fafhrd91 see the changes.

**Review by fafhrd91 — APPROVED** [review]

**Pull request merged**